sdk/storage/blobs/src/az_storage_blobs_blob_client.c

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_json_builder.h>
+#include <az_storage_blobs.h>
+
+#include <_az_cfg.h>
+
+/**
+ * @brief Maximum allowed URL size:
+ * url is expected as : [https://]{account_id}[.blob.core.windows.net]/{container}/{blob}{optional_query}
```

Comment:
blob URLs frequently DO look like this but do not have to look like this. Make sure there is NO code that has an expectation on the URL format.
URLs look different for testing, talking to private/government clouds, vanity URLs, and for scenarios using the root container.